src/libexpr/print.cc

```diff
@@ -90,5 +127,260 @@ printAttributeName(std::ostream & str, std::string_view name) {
+                    str << " elided»";
+                    if (options.ansiColors)
+                        str << ANSI_NORMAL;
+                    break;
+                }
+
+                printAttributeName(str, i.first);
+                str << " = ";
+                if (seen && seen->count(i.second)) {
+                    if (options.ansiColors)
+                        str << ANSI_MAGENTA;
+                    str << "«repeated»";
+                    if (options.ansiColors)
+                        str << ANSI_MAGENTA;
+                } else
+                    print(str, *i.second, depth + 1, seen);
+                str << "; ";
+                attrsSeen++;
+            }
+
+            str << "}";
+        } else
+            str << "{ ... }";
+
+        break;
+    }
+
+    case nList:
+        if (seen)
+            seen->insert(&v);
+
+        str << "[ ";
+        if (depth < options.maxDepth) {
+            unsigned int listItemsSeen = 0;
+            for (auto elem : v.listItems()) {
+                if (listItemsSeen >= options.maxListItems) {
+                    if (options.ansiColors)
+                        str << ANSI_FAINT;
+                    str << " «";
+                    pluralize(str, v.listSize() - listItemsSeen, "item", "items");
+                    str << " elided»";
+                    if (options.ansiColors)
+                        str << ANSI_NORMAL;
+                    break;
+                }
+
+                if (seen && seen->count(elem)) {
+                    if (options.ansiColors)
+                        str << ANSI_MAGENTA;
+                    str << "«repeated»";
+                    if (options.ansiColors)
+                        str << ANSI_NORMAL;
+                } else {
+                    if (elem) {
+                        print(str, *elem, depth + 1, seen);
+                    } else {
+                        if (options.ansiColors)
+                            str << ANSI_MAGENTA;
+                        str << "«nullptr»";
+                        if (options.ansiColors)
+                            str << ANSI_NORMAL;
+                    }
+                }
+                str << " ";
+                listItemsSeen++;
+            }
+        }
+        else
+            str << "... ";
+        str << "]";
+        break;
+
+    case nFunction:
+        if (v.isLambda()) {
+            if (options.ansiColors)
+                str << ANSI_BLUE;
+            if (v.lambda.fun) {
+                std::ostringstream s;
+                s << state.positions[v.lambda.fun->pos];
+                str << "«lambda @ " << filterANSIEscapes(s.str()) << "»";
+            } else
+                str << "«lambda»";
+            if (options.ansiColors)
+                str << ANSI_NORMAL;
+        } else if (v.isPrimOp()) {
+            if (options.ansiColors)
+                str << ANSI_MAGENTA;
```

Comment:
```suggestion
                str << ANSI_BLUE;
```

It's not that different from a `primop-app`. I think what matters is they're all functions.

Reply:
thought: If you like, you could try print which primop is applied, or (below) partially applied. (Not sure if they have names after being "detached" from their registration helper though - may have to be added, as `std::optional`?). Also if we want to print partial applications, we can actually do the same for lambdas.